Exception in thread "Thread-5" java.lang.ArrayIndexOutOfBoundsException: Index -129 out of bounds for length 128
	at cs451.PerfectLinks.send(PerfectLinks.java:86)
	at cs451.LatticeAgreement.broadcast(LatticeAgreement.java:129)
	at cs451.LatticeAgreement.lambda$startProposeThread$0(LatticeAgreement.java:106)
	at java.base/java.lang.Thread.run(Thread.java:829)
java.net.SocketException: Socket closed
	at java.base/java.net.PlainDatagramSocketImpl.receive0(Native Method)
	at java.base/java.net.AbstractPlainDatagramSocketImpl.receive(AbstractPlainDatagramSocketImpl.java:200)
	at java.base/java.net.DatagramSocket.receive(DatagramSocket.java:814)
	at cs451.UDP.lambda$startReceiveThread$1(UDP.java:102)
	at java.base/java.lang.Thread.run(Thread.java:829)
